panic: runtime error: invalid memory address or nil pointer dereference

### Error starting Nuclei:
I am getting error when starting Nuclei.
I use Nuclei in version 2.7.6 and golang in version 1.17.

```
[ERR] Could not update templates: could not read configuration file: EOF
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x10a5335]

goroutine 1 [running]:
github.com/projectdiscovery/nuclei/v2/pkg/catalog/loader.NewConfig(...)
	/root/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.7.6/pkg/catalog/loader/loader.go:79
github.com/projectdiscovery/nuclei/v2/internal/runner.(*Runner).RunEnumeration(0xc00067d810)
	/root/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.7.6/internal/runner/runner.go:380 +0xe75
main.main()
	/root/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.7.6/cmd/nuclei/main.go:90 +0x75d
```

```console
                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   2.7.6

		projectdiscovery.io

[WRN] Use with caution. You are responsible for your actions.
[WRN] Developers assume no liability and are not responsible for any misuse or damage.
Version: 2.7.6
Operative System: linux
Architecture: amd64
Go Version: go1.18.5
Compiler: gc
File "/home/user/.config/nuclei/config.yaml" Read => Ok
File "/home/user/.config/nuclei/config.yaml" Write => Ok
File "/home/user/.config/nuclei/.nuclei-ignore" Read => Ok
File "/home/user/.config/nuclei/.nuclei-ignore" Write => Ok
File "/home/user/nuclei-templates/.checksum" Read => Ko (open /home/user/nuclei-templates/.checksum: no such file or directory)
File "/home/user/nuclei-templates/.checksum" Write => Ko (open /home/user/nuclei-templates/.checksum: no such file or directory)
IPv4 connectivity to scanme.sh:80 => Ok
IPv6 connectivity to scanme.sh:80 => Ko (dial tcp6 [2400:6180:0:d0::91:1001]:80: connect: network is unreachable)
IPv4 UDP connectivity to scanme.sh:53 => Ok
```

--- a/v2/internal/runner/runner.go
+++ b/v2/internal/runner/runner.go
@@ -377,7 +377,11 @@ func (r *Runner) RunEnumeration() error {
 	}
 	executerOpts.WorkflowLoader = workflowLoader
 
-	store, err := loader.New(loader.NewConfig(r.options, r.templatesConfig, r.catalog, executerOpts))
+	templateConfig := r.templatesConfig
+	if templateConfig == nil {
+		templateConfig = &config.Config{}
+	}
+	store, err := loader.New(loader.NewConfig(r.options, templateConfig, r.catalog, executerOpts))
 	if err != nil {
 		return errors.Wrap(err, "could not load templates from config")
 	}

